driver-core/src/main/com/mongodb/event/ConnectionCheckOutFailedEvent.java

```diff
@@ -112,13 +135,35 @@ public Reason getReason() {
         return reason;
     }
 
+    /**
+     * The time it took to check out the connection.
+     * More specifically, the time elapsed between the {@link ConnectionCheckOutStartedEvent} emitted by the same checking out and this event.
+     * <p>
+     * Naturally, if a new connection was not {@linkplain ConnectionCreatedEvent created}
+     * and {@linkplain ConnectionReadyEvent established} as part of checking out,
+     * this duration is usually not greater than {@link ConnectionPoolSettings#getMaxWaitTime(TimeUnit)},
+     * but may occasionally be greater than that, because the driver does not provide hard real-time guarantees.</p>
+     * <p>
+     * This duration does not currently include the time to deliver the {@link ConnectionCheckOutStartedEvent}.
+     * Subject to change.</p>
```

Comment:
`getElapsedTime`. Just curious, in the spec, what you said about user code (which I think referred to this) seemed right. What is the reason this might change?

Reply:
🤷‍♂️ Maybe users will request for that, maybe something else I can't anticipate comes up. There is an opportunity to allow for a future change, and I prefer to take it.

Reply:
When we say "Subject to change", it seems to negate the documented behaviour immediately preceding, since users cannot rely on it. This reminded me of `@Beta`. I also thought to search for the string "Subject to change" to see if we take this approach elsewhere, and the one result was:

```
APIs marked with the `@Beta` annotation at the class or method level are subject to change.
```

For any API we do not consider Beta, I think we should either avoid documenting anything that is subject to change (in cases where the external behaviour is trivial or irrelevant to users), or commit to a behaviour. My opinion is that committing is preferable, assuming it is reasonable for a user to ask "but does this include user code"?

Reply:
> "Subject to change", it seems to negate the documented behaviour immediately preceding, since users cannot rely on it.

We could say "we don't provide any guarantees on whether this duration includes the time to deliver ...". Instead, we give users more information: we document the current behavior (which enables a user to understand why his `Thread.sleep(1_000)` in a listener does not affect the duration reported in some events), but don't guarantee it will stay the same. Also, because the current behavior is documented, users know that if it is changed, the change will be visible in the documentation.

> For any API we do not consider Beta, I think we should either avoid documenting anything that is subject to change

`@Beta` is about API changes, which is not what we are talking about here.

> My opinion is that committing is preferable, assuming it is reasonable for a user to ask "but does this include user code"

1. This PR implements the spec requirement _"The driver MUST document this behavior as well as explicitly warn users that the behavior may change in the future"_. Are you require the spec to be re-considered? If yes, is this concern a PR blocker? Note that if a DRIVERS ticket is created, and the spec is changed as a result, incorporating such a change in the Java driver will not break compatibility.
2. _assuming it is reasonable for a user to ask "but does this include user code"_ - I don't understand this part. Could you explain it?

Reply:
Described behaviour is rightly taken by users to be part of the public API. Some public behaviour might be tentative ("subject to change"), but my view is that all such instances in the driver should be marked with `@Beta`, and that this is done with the expectation that we think the present design is right, and will "soon" be removed from Beta. I do think it is a mistake for any non-beta spec to require drivers to mark public behaviour as tentative.

The original wording was describing public behaviour, however, I believe you are proposing that we might instead state this behaviour but clarify that it is non-public. I don't think we (or others) should describe implementation details, especially without a compelling reason. Otherwise, we might include many internal hints and tips in our docs.

So:

1. If public behaviour is "subject to change", it should be marked Beta, which is not applicable here.
2. Non-public behaviour is inherently "subject to change", but I think we should not mention internal behaviour.

My opinion is that you made the right call in excluding user code, and that we should publicly commit to that behaviour. I do not mind merging and changing this later via tickets, if that is more convenient.